### PR TITLE
Add resolvename subdomains

### LINKED ISSUE #911: [DOTNAV] Add resolve subdomains option to resolvename
Add a boolean option to `resolvename` so it also shows the different subdomains of a name.

Example:

`resolvename navcoin.nav` Output:

```
{
  "_expiry": "6978256",
  "_key": "acce5028f8e7c95a4e3f676e76b3fddb21db12b42de9182aa0cd047d282e35ff7f64b4cf12d02897d79f558e94c6cb14",
  "ip": "127.0.0.1"
}
```

`resolvename Navcoin.nav true` Output:

```
{
  "_expiry": "6978256",
  "_key": "acce5028f8e7c95a4e3f676e76b3fddb21db12b42de9182aa0cd047d282e35ff7f64b4cf12d02897d79f558e94c6cb14",
  ".": {
    "ip": "127.0.0.1"
  },
  "www": {
    "ip": "8.8.8.8"
  }
}
```

### PR DESCRIPTION
**Pull request by mxaddict**
closes #911

image

image

### CODE REVIEW

**Comment by navbuilder** [comment]
A new build of 302395cddda7c941e22bef6ce7201a5a9b1d61ad has completed succesfully!
Binaries available at https://build.nav.community/binaries/add-resolvename-subdomains

**Review by aguycalled — COMMENTED** [review]
I would say to include a check so If the client tries to resolve a name with subdomain like subdomain.name.nav, the resolve subdomain option can't be specified

**Comment by mxaddict** [comment]
> I would say to include a check so If the client tries to resolve a name with subdomain like subdomain.name.nav, the resolve subdomain option can't be specified

Alright, I'll add it now, should it not allow it? Or just ignore the second param?

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1358,12 +1365,38 @@ UniValue resolvename(const UniValue& params, bool fHelp)
...
     auto mapData = DotNav::Consolidate(data, chainActive.Tip()->nHeight, subdomain);
 
     for (auto &it: mapData) {
-        ret.pushKV(it.first, it.second);
+        if (it.first.substr(0,1) == "_") {
+            ret.pushKV(it.first, it.second);
+        }
+    }
+
+    if (getSubdomains && subdomain != "") {
```
Comment (mxaddict) [inline src/wallet/rpcwallet.cpp:1377]: @aguycalled I added this check, should be good?

**Comment by chasingkirkjufell** [comment]
i thought that each dotnav can only have one subdomain?  so there wouldn't be subdomains

**Comment by navbuilder** [comment]
A new build of b8b7ff3fb6d8f16424988abe42604cd8ca5ee9a9 has completed succesfully!
Binaries available at https://build.nav.community/binaries/add-resolvename-subdomains

**Comment by mxaddict** [comment]
Yes, that is why we needed to add the check logic.

**Review by aguycalled — APPROVED** [review]

**Comment by chasingkirkjufell** [comment]
I'm can't resolve subdomain with true unless direction resolving subdomain. did i do it wrong?
image

**Comment by mxaddict** [comment]
> i thought that each dotnav can only have one subdomain? so there wouldn't be subdomains

Yes, we can't have second level subdomains, that is why we just ignore the subdomain request on a subdomain

**Comment by chasingkirkjufell** [comment]
In my screenshot I was trying to resolve first level subdomain with true on the original domain but it didnt work. Can only resolve subdomain directly.

**Comment by mxaddict** [comment]
> 

I'll double check

**Comment by mxaddict** [comment]
@chasingkirkjufell I've fixed the issue that you mentioned

**Comment by navbuilder** [comment]
A new build of 555fa81aebc3516e48fd176b9fd1945ea0bc8a69 has completed succesfully!
Binaries available at https://build.nav.community/binaries/add-resolvename-subdomains

**Review by chasingkirkjufell — APPROVED** [review]
testes gitian build on windows 10

**Pull request merged**